the-collab-lab/tcl-36-smart-shopping-list
# User Story 7 Welcoming prompt

## Description

- Created a Welcome component that will render on the /listView page when a user has 0 items in their list.
- The current design is based on a card layout that features a button, prompting users to add items to their empty list.
- Added Tailwind CSS, Craco, PostCSS and related dependencies to make styling components easier and added sample border, color, and hover designs to present to get feedback from the team.

## Related Issue

Closes #7 

## Acceptance Criteria

* The list view, when there are no items to display, should show a prompt (e.g., a button) for the user to add their first item
Notes:
* The wireframe is a little short on the "welcoming" part of the story, but otherwise is functionally complete for the purposes of the AC. https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/list-empty.png



## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|  ✓  | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before
![listView](https://user-images.githubusercontent.com/89806097/152217968-d4ea9a6e-b642-4acb-9a23-6d7f6913492e.png)


### After

![listView](https://user-images.githubusercontent.com/89806097/152218032-cbee558f-19fc-4b81-88eb-a49152f735bd.png)

## Testing Steps / QA Criteria

- From your terminal fetch branch information from the remote repository using `git fetch origin ld-dj-make-app-welcoming`, and then checkout our branch with `git checkout ld-dj-make-app-welcoming`
- Install tailwind dependencies using `npm install` in your terminal
- Clear your local storage in the browser console using `localStorage.clear()`
- Navigate to localhost:3000/listView
- Does this UI make you want to add an item?
- Click the link to add an item and confirm it you are directed to /addItemView
